docs/reference/query-dsl.asciidoc

```diff
@@ -25,6 +25,27 @@ or to alter their behaviour (such as the
 
 Query clauses behave differently depending on whether they are used in
 <<query-filter-context,query context or filter context>>.
+
+[[query-dsl-allow-expensive-queries]]
+Allow expensive queries::
+Execution of certain types of queries have usually slow performance, which can affect the cluster performance.
+Those queries can be categorised as follows:
+* Queries that need to do linear scans to identify matches:
+** <<query-dsl-script-query, `script queries`>>
+** <<query-dsl-script-score-query, `script score queries`>>
```

Comment:
I was expecting this one to be fine since it finds matches using another query?

Reply:
It was a suggestion by @jimczi to also include those, since I guess the custom score calculation could decrease performance ?